lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -393,6 +393,20 @@ def connect(
         except CalledProcessError as e:
             raise SnapError("Could not {} for snap [{}]: {}".format(_cmd, self._name, e.stderr))
 
+    def hold(self, duration: Optional[str] = None) -> None:
+        """Add a refresh hold to a snap.
+
+        Args:
+            duration (str): (optional) duration of the hold.
+                Default `None`
+        """
+        args = f"--hold={duration}" if duration else "--hold"
```

Comment:
We're not doing any validation on the format of `duration`, what happens if I just call `snap.hold(duration="FOOBAR")`?